### PR TITLE
Removed deprecated VariantContextWriterFactory

### PR DESCRIPTION
**Pull request by magicDGS**
### Description

Removed ancient factory for create `VariantContextWriter`. Found in #707.

### Checklist

- [x] Code compiles correctly
- [ ] New tests covering changes and new functionality
- [x] All tests passing
- [ ] Extended the README / documentation, if necessary
- [ ] Is not backward compatible (breaks binary or source compatibility)

### CODE REVIEW

File: src/main/java/htsjdk/variant/variantcontext/writer/VariantContextWriterFactory.java [hunk 1]
```diff
@@ -1,282 +0,0 @@
...
-    }
-
-    public static boolean isCompressedVcf(final File location) {
-        if (location == null)
-            return false;
-
-        return AbstractFeatureReader.hasBlockCompressedExtension(location);
-    }
-
-    public static VariantContextWriter sortOnTheFly(final VariantContextWriter innerWriter, final int maxCachingStartDistance) {
```
Comment (ronlevine) [inline src/main/java/htsjdk/variant/variantcontext/writer/VariantContextWriterFactory.java]: Does `sortOnTheFly` need to be added to `VariantContextWriterBuilder` ?

Comment (ronlevine) [inline src/main/java/htsjdk/variant/variantcontext/writer/VariantContextWriterFactory.java]: @magicDGS I agree with you, no need to add it. The builder pattern should not have any static methods plus it the a VERY thin wrapper around SortingVariantContextWriter.

**Comment by yfarjoun** [comment]
PR in picard ready for this https://github.com/broadinstitute/picard/pull/710 (tests pass)

**Comment by yfarjoun** [comment]
👍

**Pull request merged**